Gems/Atom/RPI/Code/Tests/Material/MaterialTypeAssetTests.cpp

```diff
@@ -490,6 +499,32 @@ namespace UnitTest
         });
     }
 
+    TEST_F(MaterialTypeAssetTests, Error_InvalidMaterialVersionUpdate)
+    {
+        Data::Asset<MaterialTypeAsset> materialTypeAsset;
+
+        Data::AssetId assetId(Uuid::CreateRandom());
+
+        MaterialTypeAssetCreator materialTypeCreator;
+        materialTypeCreator.Begin(assetId);
+
+        // Invalid version updates
+        MaterialVersionUpdate versionUpdate(2);
+        versionUpdate.AddAction(MaterialVersionUpdate::Action(AZ::Name{ "rename" }, {
+            { Name{ "from" }, Name{ "EnableSpecialPassPrev" } },
+            { Name{ "to" }, Name{ "InvalidPropertyName" } } }));
+        materialTypeCreator.SetVersion(versionUpdate.GetVersion());
+        materialTypeCreator.AddVersionUpdate(versionUpdate.GetVersion(), versionUpdate);
+        materialTypeCreator.AddShader(m_testShaderAsset);
+
+        materialTypeCreator.BeginMaterialProperty(Name{ "EnableSpecialPass" }, MaterialPropertyDataType::Bool);
+        materialTypeCreator.EndMaterialProperty();
+
+        AZ_TEST_START_ASSERTTEST;
```

Comment:
This is fine, but just FYI check out ErrorMessageFinder if you haven't seen that before, it's even better.